frontend/src/components/Sidebar/Sidebar.tsx

```diff
@@ -33,7 +36,10 @@ export const Sidebar: React.FC<Props> = ({ schedule, major }) => {
     );
   }
 
-  const warnings = produceRequirementGroupWarning(schedule, major);
+  const warnings = produceRequirementGroupWarning(
+    JSON.parse(JSON.stringify(schedule)),
+    major
+  ); // deep copy of schedule
```

Comment:
why deep copy of schedule? If produceRequirementGroupWarning somehow mutates, PLEASE let me know (I need to change that). If it's because the produced warnings contain references to things in the schedule, also please let me know (I should change that too).

Reply:
I was running in to trouble with this because since `schedule` is from redux, it cannot be directly mutated. If I remember correctly it was caused by a `.sort` call.

Reply:
Thanks, I'll look into it. It shouldn't be mutating, so that sounds like an error.